examples/widgets/annotated_cursor.py

```diff
@@ -0,0 +1,345 @@
+            # same position (mouse moves slightly between two plot points)
+            # can be skipped
+            self.lastdrawnplotpoint = plotpoint
+        # otherwise, make text invisible
+        else:
+            self.text.set_visible(False)
+
+        # Draw changes. Canot use _update method of baseclass,
+        # because it would first restore the backround, which
+        # is done already and is not necessary.
+        if self.useblit:
+            self.ax.draw_artist(self.text)
+            self.canvas.blit(self.ax.bbox)
+        else:
+            # If blitting is deactivated, the overwritten _update call made
+            # by the base class immedeately returned.
+            # We still have to draw the changes.
+            self.canvas.draw_idle()
+
+    def setpos(self, xpos, ypos):
+        """
+        Finds the coordinates, which have to be shown in text.
+
+        The behaviour depends on the *dataaxis* attribute. Function looks
+        up the matching plot coordinate for the given mouse position.
+
+        Parameters
+        ----------
+        xpos : float
+            The current x position of the cursor in data coordinates.
+            Important if *dataaxis* is set to 'x'.
+        ypos : float
+            The current y position of the cursor in data coordinates.
+            Important if *dataaxis* is set to 'y'.
+
+        Returns
+        -------
+        ret : {2D array-like, None} The coordinates which should be displayed.
+            *None* is the fallback value.
+        """
+
+        # Get plot line data
+        xdata = self.line.get_xdata()
+        ydata = self.line.get_ydata()
+
+        # The dataaxis attribute decides, in which axis we look up which cursor
+        # coordinate.
+        if self.dataaxis == 'x':
+            pos = xpos
+            data = xdata
+            lim = self.ax.get_xlim()
+        elif self.dataaxis == 'y':
+            pos = ypos
+            data = ydata
+            lim = self.ax.get_ylim()
+        else:
+            raise ValueError(
+                "The data axis specifier {} should be 'x' or 'y'"
+                .format(self.dataaxis))
+
+        # If position is valid
+        if pos is not None:
+            # And in valid plot data range
+            if pos >= lim[0] and pos <= lim[-1]:
+                # Convert given positon to numpy array,
+                # so numpy function can be used.
+                findme = np.array([pos])
+                # Find closest x value in sorted x vector.
+                # This is the code line,
+                # which requires the plotted data to be sorted.
+                index = np.searchsorted(data, findme)
+                # Return none, if this index is out of range.
+                if (index < 0) or (index >= len(data)):
+                    return None
+                # Return plot point as tuple.
+                return (xdata[index][0], ydata[index][0])
```

Comment:
```suggestion
        # If position is valid and in valid plot data range.
        if pos is not None and lim[0] <= pos <= lim[-1]:
            # Find closest x value in sorted x vector.
            # This requires the plotted data to be sorted.
            index = np.searchsorted(data, pos)
            # Return none, if this index is out of range.
            if index < 0 or index >= len(data):
                return None
            # Return plot point as tuple.
            return (xdata[index][0], ydata[index][0])
```

Reply:
Commit d027645ba81649eccaa127a739235cae5efc8c0a fixes a bug introduced with this change: Had to remove [0] in the returned tuple

Reply:
Right, I forgot to remove those.